src/stream.cpp

```diff
@@ -1629,7 +1634,12 @@ namespace stream {
       auto session = (session_t *) channel_data;
 
       auto sequenceNumber = session->audio.sequenceNumber;
-      auto timestamp = session->audio.timestamp;
+      // Audio timestamps are in milliseconds and should be AudioPacketDuration (5ms or 10ms) apart
+      auto timestamp = static_cast<std::uint32_t>(
+        std::chrono::duration_cast<std::chrono::microseconds>(
+          std::chrono::steady_clock::now() - session->audio.timestamp_epoch
```

Comment:
I think it's possible to get capture time from the OS too like we do with video frames. `IAudioCaptureClient::GetBuffer()` takes an optional `pu64QPCPosition` parameter that gives you the timestamp of capture.